pkg/sabakan-cryptsetup/cmd/driver.go

```diff
@@ -152,7 +156,18 @@ func (d *Driver) setupDisk(ctx context.Context, disk Disk, tpmKek []byte) error
 		})
 		return d.formatDisk(ctx, disk, f, tpmKek)
 	}
-	return err
+
+	log.Error("failed to retrieve key from sabakan", map[string]interface{}{
+		log.FnError: err,
+		"disk":      disk.Name(),
+		"try":       retries + 1,
+	})
+	if retries == maxRetry {
+		return err
+	}
+	retries++
```

Comment:
`retries++` should be before `if retries == maxRetry` block or else retry happens (maxRetry + 1) times.

Reply:
No.  Retries can happen exactly at most `maxRetry` times.
Note that the first try is *not* a retry.